src/main/java/uk/gov/hmcts/ccd/domain/service/stdapi/CallbackInvoker.java

```diff
@@ -167,7 +173,8 @@ private void validateAndSetFromAboutToStartCallback(CaseTypeDefinition caseTypeD
         callbackService.validateCallbackErrorsAndWarnings(callbackResponse, ignoreWarning);
 
         if (callbackResponse.getData() != null) {
-            validateAndSetData(caseTypeDefinition, caseDetails, callbackResponse.getData());
+            validateAndSetData(caseTypeDefinition, caseDetails, callbackResponse.getData(),
```

Comment:
can we avoid these flags here and define two delegate methods 

- validateAndSetData(caseTypeDefinition, caseDetails, callbackResponse.getData())
- validateAndSetDataWithSearcCriteria(caseTypeDefinition, caseDetails, callbackResponse.getData())

Both of them internally pass a flag to core `validateAndSetData` method ..

Reply:
Fixed in 9f001db